File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    let isVowel = new Set("aeiouAEIOU");
    let vowels = [];
    for (let i of s) {
        if (isVowel.has(i)) {
            vowels.push(i);
        }
    }
    vowels=vowels.reverse();
    let index = 0;
    let result = "";
    for (let i of s) {
        if (isVowel.has(i)) {
            result += vowels[index];
            index++
        } 
        else {
            result += i;
        }
    }

    return result;
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        set <char> isVowel={'a','e','i','o','u','A','E','I','O','U'};
        int i=0;
        int j=s.size()-1;
        while(i<j){
            while(i<j && isVowel.count(s[i])==0){
                i++;
            }
            while(i<j && isVowel.count(s[j])==0){
                j--;
            }
            swap(s[i],s[j]);
            i++;
            j--;
        }
        return s;
    }
};